mod-tool/src/main/scala/at/forsyte/apalache/tla/tooling/opt/TraceeCmd.scala

```diff
@@ -0,0 +1,92 @@
+package at.forsyte.apalache.tla.tooling.opt
+
+import at.forsyte.apalache.infra.passes.PassChainExecutor
+import at.forsyte.apalache.infra.passes.options._
+import at.forsyte.apalache.io.json.JsonDeserializationError
+import at.forsyte.apalache.io.json.impl.{UJsonRep, UJsonScalaFactory}
+import at.forsyte.apalache.tla.bmcmt.config.TraceeModule
+import org.backuity.clist._
+
+import java.io.File
+import scala.util.{Failure, Success, Try}
+
+/**
+ * This command initiates the 'tracee' command line.
+ *
+ * @author
+ *   Jure Kukovec
+ */
+class TraceeCmd(name: String = "tracee", description: String = "Evaluate expressions over a trace.")
+    extends CheckCmd(name, description) {
+
+  var trace: File = arg[File](description = "a file containing an ITF trace. Must also define --expressions.")
+
+  var expressions: List[String] =
+    arg[List[String]](name = "expressions",
+        description = "TLA+ expressions to be evaluated over a given trace. Must also define --trace.")
+
+  // If we set the --length inside collectTuningOptions it gets overwritten later.
+  // therefore, we store it locally, and only update the options right before calling run()
+  private var persistentLen: Int = 0
+
+  def getLenFromFile(file: File): Int = Try(UJsonRep(ujson.read(file))) match {
+    case Success(json) =>
+      // ITF case
+      val statesOpt = json.getFieldOpt("states").map(UJsonScalaFactory.asSeq)
+      statesOpt match {
```

Comment:
You already have a trace reader in `UJsonTraceReader`, in particular the dropping of `Cinit` and `Inv` is duplicated.

Reply:
That's true for reading, and it should be used thhere, but running the entire `convert`, just to read the length of a json array (the contents of which do not need to be converted at this point, and cannot be made persistent) seems wasteful. The length of the unconverted array should be the same as that of the converted one.

Reply:
Looking up JSON fields in a `Cmd` feels wrong, why not move this to `UJsonTraceReader`?

Reply:
That's probably not a bad idea.